src/Bling.js

```diff
@@ -378,6 +394,11 @@ class Bling extends Component {
             : Bling._config.viewableThreshold;
     }
 
+    componentWillMount() {
+        const {npa} = this.props;
+        this.handleSetNpaFlag(npa);
```

Comment:
`componentWillMount` happens on the server (for people using react-ssr). i wouldnt recommend putting this method here. 
Maybe it would be better in `defineSlot()`? 
```
defineSlot() {
    const { adUnitPath, outOfPage, shouldUseNonPersonalizedAds } = this.props;
    const divId = this._divId;
    const slotSize = this.getSlotSize();

    if (!this._adSlot) {
        if (outOfPage) {
            this._adSlot = Bling._adManager.googletag.defineOutOfPageSlot(
                adUnitPath,
                divId
            );
        } else {
            this._adSlot = Bling._adManager.googletag.defineSlot(
                adUnitPath,
                slotSize || [],
                divId
            );
        }
    }
    // 0 for personalized (default) 1 for non-personalized 
    Bling._adManager.googletag.pubads().setRequestNonPersonalizedAds(shouldUseNonPersonalizedAds ? 1 : 0);

    this.configureSlot(this._adSlot);
}

```

Reply:
It is a great hint. Thanks. 
I've followed it and removed `componentWillMount` completelly.